### PR TITLE
Move curand handle out of CuRand class and into CuDevice. 

### PR DESCRIPTION
**Pull request by luitjens**
This matches the behavior of other CUDA libraries like cublas, cufft, and cusparse.
This change is necessary to avoid redundant creation/destroying of the
random generator every-time someone creates a CuRand object.  In my measurements
I noticed that a single instantiation of CuRand<> could take up to 3 ms.

The CuRand object is now a thin wrapper around the generators and should
have very little overhead.

### CODE REVIEW

**Comment by danpovey** [comment]
Cool.  I assume you made sure the tests ran with GPUs?

**Comment by luitjens** [comment]
I ran a single test on my system and it worked fine.  That isn't very
extensive.  Do you have internal tests that stress GPUs?

**Comment by luitjens** [comment]
Pushed a new patch.  make test works in src with cuda enabled.

File: src/cudamatrix/cu-device.h [hunk 1]
```diff
@@ -85,10 +85,12 @@ class CuDevice {
   inline curandGenerator_t GetCurandHandle() { return curand_handle_; }
 
   inline void SeedGpu() {
-		// To get same random sequence, call srand() before the method is invoked,
-		CURAND_SAFE_CALL(curandSetPseudoRandomGeneratorSeed(
-          curand_handle_, RandInt(128, RAND_MAX)));
-		CURAND_SAFE_CALL(curandSetGeneratorOffset(curand_handle_, 0));
+    if(CuDevice::Instantiate().Enabled()) {
```
Comment (danpovey) [inline src/cudamatrix/cu-device.h]: sorry very small comment, but space after 'if' please.

Comment (danpovey) [reply src/cudamatrix/cu-device.h]: actually its OK, I can change it from github directly.

**Comment by danpovey** [comment]
will  merge once checks complete

**Comment by luitjens** [comment]
thanks!

**Pull request merged**